testing/python/fixtures.py

```diff
@@ -3950,3 +3950,27 @@ def fix():
 
     with pytest.raises(pytest.fail.Exception):
         assert fix() == 1
+
+
+# https://github.com/pytest-dev/pytest/issues/5036
+def test_fixture_param_shadowing(testdir):
+    testdir.makepyfile(
+        """
+        import pytest
+
+        @pytest.fixture(params=['a', 'b'])
+        def argroot(request):
+            return request.param
+
+        @pytest.fixture
+        def arg(argroot):
+            return argroot
+
+        @pytest.mark.parametrize("arg", [1])
+        def test_simple(arg, request):
+            assert arg == 1
+    """
+    )
+    result = testdir.runpytest()
+    # Only one test should have run
+    result.assert_outcomes(passed=1)
```

Comment:
I suggest to pass `-v` so the actual expectation is clearer. 👍 

```suggestion
    result = testdir.runpytest("-v")
    result.stdout.fnmatch_lines(["::test_simple[1]"])
```

Reply:
My bad, this should be:

```python
result.stdout.fnmatch_lines(["*::test_direct[1]*"])
```